src/Ombi.Core/Engine/TvRequestEngine.cs

```diff
@@ -896,6 +905,52 @@ private async Task CheckForSubscription(HideResult shouldHide, List<ChildRequest
             }
         }
 
+        private class EpisodeKey
+        {
+            public int SeasonNumber;
+            public int EpisodeNumber;
+        }
+
+        private void CheckForPlayed(HideResult shouldHide, List<ChildRequests> childRequests)
+        {
+            var theMovieDbIds = childRequests.Select(x => x.Id);
+            foreach (var request in childRequests)
+            {
+                var requestedEpisodes = getEpisodesKeys(request);
+
+                var playedEpisodes = _userPlayedEpisodeRepository
+                    .GetAll()
+                    .Where(x => x.TheMovieDbId == request.Id && x.UserId == request.RequestedUserId)
+                    .AsEnumerable()
+                    .Join(requestedEpisodes,
+                        played => new { played.SeasonNumber, played.EpisodeNumber },
+                        requested => new { requested.SeasonNumber, requested.EpisodeNumber },
+                        (played, requested) => new { played });
```

Comment:
The above, in the join we are not using the result from `requested`, so you can remove that

Reply:
I'm not sure what you mean but this is probably because I'm still discovering the Join feature.
Do you mean this?
```
.Join(requestedEpisodes,
                        played => new { played.SeasonNumber, played.EpisodeNumber },
                        requested => new { requested.SeasonNumber, requested.EpisodeNumber },
                        (played) => new { played });
```
If so, this gives me a syntax error (CS0411). Am I missing something?

If you mean that the join itself is not necessary, I use it to additionally filter the result of my query so that it only returns episodes that are in requestedEpisodes. This is what I tried to explain on Discord earlier.